applications/views/applications.py

```
@@ -409,6 +410,9 @@ def put(self, request, pk):
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
+        if not can_set_status(application, data["status"]):
```

I'd raise a ValidationError here instead of returning a JsonResponse :)

---

Added! :)